From aa74f88c3e1e8eeb7a946585cbad1043be5c3253 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:53 -0400
Subject: [powerpc] kvm: Copy the pvr value after memset

Message-id: <1410545655-205645-65-git-send-email-dzickus@redhat.com>
Patchwork-id: 94109
O-Subject: [RHEL7 PATCH 064/626] powerpc/kvm: Copy the pvr value after memset
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 87916442bd0ac0aaabcca4f7fcd7eb3d8bef0d34
Author: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>
Date:   Thu Aug 22 17:08:39 2013 +0530

    powerpc/kvm: Copy the pvr value after memset

    Otherwise we would clear the pvr value

    Signed-off-by: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>
    Signed-off-by: Alexander Graf <agraf@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kvm/book3s_hv.c b/arch/powerpc/kvm/book3s_hv.c
index 18cab29..595acce 100644
--- a/arch/powerpc/kvm/book3s_hv.c
+++ b/arch/powerpc/kvm/book3s_hv.c
@@ -680,13 +680,12 @@ static int kvmppc_handle_exit(struct kvm_run *run, struct kvm_vcpu *vcpu,
 }
 
 int kvm_arch_vcpu_ioctl_get_sregs(struct kvm_vcpu *vcpu,
-                                  struct kvm_sregs *sregs)
+      struct kvm_sregs *sregs)
 {
  int i;
 
- sregs->pvr = vcpu->arch.pvr;
-
  memset(sregs, 0, sizeof(struct kvm_sregs));
+ sregs->pvr = vcpu->arch.pvr;
  for (i = 0; i < vcpu->arch.slb_max; i++) {
   sregs->u.s.ppc64.slb[i].slbe = vcpu->arch.slb[i].orige;
   sregs->u.s.ppc64.slb[i].slbv = vcpu->arch.slb[i].origv;
@@ -696,7 +695,7 @@ int kvm_arch_vcpu_ioctl_get_sregs(struct kvm_vcpu *vcpu,
 }
 
 int kvm_arch_vcpu_ioctl_set_sregs(struct kvm_vcpu *vcpu,
-                                  struct kvm_sregs *sregs)
+      struct kvm_sregs *sregs)
 {
  int i, j;
 
-- 
1.7.1